mobile-ui/src/main/java/org/sagebionetworks/research/mobile_ui/perform_task/PerformTaskFragment.java

```diff
@@ -218,6 +247,11 @@ void afterLastStep() {
         String sharedPreferencesKey = this.taskView.getIdentifier();
         SharedPreferences sharedPreferences = getContext().getSharedPreferences(sharedPreferencesKey,
                 Context.MODE_PRIVATE);
+
+        sharedPreferences.edit().putLong(LAST_RUN_KEY, Instant.now().toEpochMilli()).apply();
```

Comment:
These two shared preferences edits look like a merge issue to me.

checkExitListener should go below the sharedpreferences.edit below, as the last line of this method